_posts/2017-06-23-twitter-digest.html

```diff
@@ -0,0 +1,72 @@
+</p>
+
+<h2>Twitter Digest, explained</h2>
+<p>
+	We focused our experiment on one of the most popular streams of information, Twitter. By combining Twitter and a Chrome extension, users were able to group content around a word of their chosing.
+</p>
+<p>
+	Any tweet in the user's timeline containing the keyword is then 'extracted' from it and appears in a cluster at the top of the timeline. It is then up to them to choose to explore the tweets in the cluster or scroll past them.
+</p>
+<p>
+	Twitter Digest is merely an attempt to reduce the scattered noise in a feed by condensing and surfacing tweets. It differs from other filters or "spoiler alert" type of extensions as it doesn't hide or dismiss any content but offers a new reading mode.
+</p>
+
+<img src="/assets/uploads/2017/06/twitter_digest.png" alt="Twitter Digest in situ." />
+
+<h2>Technical details & challenges</h2>
+<p>
+	Twitter Digest comprises of a node server, which handles authentication and keyword requests, and a Chrome extension which interacts with the client side (in this case the DOM of <a href="https://twitter.com/" target="_blank">twitter.com</a>).
+</p>
+<p>
+	The source code is available on <a href="https://github.com/ftlabs/Twitter-digest" target="_blank">GitHub</a>.
+</p>
+<p>
+	In the early stages of the prototype, Twitter Digest didn't allow for a choice of keyword and required a JSON object to be pasted in the popup of the extension to be processed. It can now handle switching between accounts, signing in with Twitter and toggling the digest on and off. We have made a few discoveries throughout the process.
+</p>
+
+	<h3>Twitter API</h3>
+<p>
+	It took a few tries to find the correct endpoint of the Twitter API for the digest. After considering <a href="https://dev.twitter.com/rest/reference/get/statuses/user_timeline" target="_blank">user_timeline</a>, we settled on <a href="https://dev.twitter.com/rest/reference/get/statuses/home_timeline" target="_blank">home_timeline</a> which offers the most accurate depiction of what is in a user's feed, although it excludes tweets followees have liked and the 'In case you missed it section'.
+</p>
+<p>
+	OAuth, as always, has proved tricky, especially with the added layer of communication between the extension and the Node server. Ensure you have all the necessary callbacks in place, and above all that the callback URL set when creating the Twitter application matches the one you've implemented. (This needs to be amended when switching from a local environment to a live server).
```

Comment:
This is a weird sentence from "Ensure" onwards. It's as if we're walking a developer through making their own Twitter client, which I don't think fits well in this context.